### ISSUE TITLE
Writing to a new zip archive member without explicitly creating it raises MNU

### ISSUE THREAD

**Issue by theseion** [guillep/pharo-core#4897]
See #2227.

The following should not fail with an MNU. The archive member should either be created implicitly or a proper exception should be raised:

```Smalltalk
| zipFile zip |
zipFile := FileSystem memory workingDirectory / 'test.zip'.
zip := (FileSystem zip: zipFile) open.
zip / 'sample.txt'
  writeStreamDo: [ :stream |
  stream nextPutAll: 'test' ];
  yourself.
zip workingDirectory files first readStreamDo: [ :s | s contents ]
```

**Comment by myroslavarm** [comment]
is this still relevant?

**Comment by guillep** [comment]
@myroslavarm have you tried the code?

With the example above you should be able to decide if this still happens in Pharo9 or not for example.

**Comment by theseion** [comment]
I'm pretty sure, it is. Unless someone suddenly came up with a new implementation, which I doubt.

**Comment by myroslavarm** [comment]
@guillep i don't know how to add a zip file to the memory filesystem

**Comment by theseion** [comment]
You don't have to add a Zip file. The example above already creates one (just tested in Pharo 8).

**Comment by myroslavarm** [comment]
well, i just tested it in pharo 9 and i have the following error:
`#uncompressedSize was sent to nil`
is this what you meant by the MNU error?

**Comment by theseion** [comment]
Yes, exactly :) "MNU": `MessageNotUnderstood`, which is the exception signalled in this case. Sorry about the confusion.

**Comment by guillep** [comment]
Thanks all!
